pyam/core.py

```diff
@@ -489,34 +489,69 @@ def pivot_table(self, index, columns, values='value',
         df = df.unstack(level=columns, fill_value=fill_value)
         return df
 
-    def interpolate(self, time):
-        """Interpolate missing values in timeseries (linear interpolation)
+    def interpolate(self, time, **kwargs):
+        """Interpolate missing values in timeseries (linear interpolation by
+        default). Internally, :meth:`pandas.DataFrame.interpolate` is used.
 
         Parameters
         ----------
-        time : int or datetime
+        time : int or datetime, or list-like thereof
              Year or :class:`datetime.datetime` to be interpolated.
              This must match the datetime/year format of `self`.
+        inplace : bool, default False
+            if True, do operation inplace and return None
+        kwargs
+            passed to :meth:`pandas.DataFrame.interpolate`
         """
-        if self.time_col == 'year' and not isinstance(time, int):
+        ##
+        # TODO remove, and add kwarg inplace=False in next release
+        if 'inplace' not in kwargs:
```

Comment:
if suggestion above to add kwarg to signature is implemented
```suggestion
        if 'inplace' is None:
```